sru/modules.py

```diff
@@ -123,36 +122,35 @@ def __init__(self,
         self.weight_c_init = weight_c_init
 
         # projection dimension
-        self.projection_size = 0
-        if n_proj > 0 and n_proj < self.input_size and n_proj < self.output_size:
-            self.projection_size = n_proj
+        self.projection_size = projection_size
 
         # number of sub-matrices used in SRU
         self.num_matrices = 3
         if has_skip_term and self.input_size != self.output_size:
             self.num_matrices = 4
 
-        # make parameters
-        if self.custom_m is None:
+        if custom_m is None:
+            # create an appropriate custom_m, depending on whether we are using projection or not
             if self.projection_size == 0:
-                self.weight_proj = None
-                self.weight = nn.Parameter(torch.Tensor(
-                    input_size,
-                    self.output_size * self.num_matrices
-                ))
+                # use an nn.Linear
+                custom_m = nn.Linear(
+                    input_size, self.output_size * self.num_matrices, bias=False)
             else:
-                self.weight_proj = nn.Parameter(torch.Tensor(input_size, self.projection_size))
-                self.weight = nn.Parameter(torch.Tensor(
-                    self.projection_size,
-                    self.output_size * self.num_matrices
-                ))
+                # use a Sequential[nn.Linear, nn.Linear]
+                custom_m = nn.Sequential(
+                    nn.Linear(input_size, self.projection_size, bias=False),
+                    nn.Linear(
+                        self.projection_size, self.output_size * self.num_matrices, bias=False),
+                )
+        self.custom_m = custom_m
```

Comment:
Since we don't need to worry about backward compatibility. Shall we use a more readable / interpretable name than `custom_m`?

Reply:
good idea :)

Reply:
`W`? `w`? `h1`? `hW`?

Reply:
(currently I'm semi leaning towards `UV_net`)

Reply:
These sound good to me:
`transform_module`,
`input_projection_module`, 
`input_transform_module`

Reply:
ok, let's go with `transform_module`, since it is the shortest. I'll make the change.